### PR TITLE
Add documentation for all services

### PR DESCRIPTION
**Pull request by Malaber**

### CODE REVIEW

File: app/services/playoff_stage_service.rb [hunk 1]
```diff
@@ -1,32 +1,52 @@
...
+  #
+  # @param stage_count [Integer] number of stages to generate
+  # @return [Array] the generated stages
   def self.generate_stages_with_empty_matches(stage_count)
     empty_stages = []
     stage_count.times do |i|
       stage = Stage.new level: i, matches: generate_empty_matches(2**i)
       empty_stages << stage
     end
+    # as we are generating the stages in the wrong order (starting with the lowest number of matches (which is the final stage)) they need to be reversed
```
[the remark this answers is not in the document]
Comment (Malaber) [reply app/services/playoff_stage_service.rb]: I hate you hound

File: app/services/playoff_stage_service.rb [hunk 2]
```diff
@@ -1,32 +1,52 @@
 # frozen_string_literal: true
 
 class PlayoffStageService
+
```
[the remark this answers is not in the document]
Comment (Malaber) [reply app/services/playoff_stage_service.rb]: Please don't annoy me!

File: app/services/match_service.rb [hunk 3]
```diff
@@ -1,22 +1,31 @@
...
     end
 
+    # normal_games = number of matches with two teams attending
+    # needed_games = number of matches to generate in total for the given number of teams
     if Utils.po2?(teams.size)
+      # if amount of teams is power of two, all matches are with two teams
       normal_games = teams.size / 2
       needed_games = normal_games
     else
+      # if amount of teams isn't a power of two we need to "kick out" enough teams to get to a power of two in the next round
```
[the remark this answers is not in the document]
Comment (Malaber) [reply app/services/match_service.rb]: This needs to stop...

**Review by Thor77 — APPROVED** [review]
LGTM

**Pull request merged**